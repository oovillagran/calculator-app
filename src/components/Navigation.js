import { Link } from 'react-router-dom';

export default function Navigation() {
  return (
    <>
      <nav className="nav-bar">
        <ul className="nav-container">
          <li className="nav-item">
            <Link to="/">Home</Link>
          </li>
          <li className="nav-item">
            <Link to="/calculator">Calculator</Link>
          </li>
          <li className="nav-item">
            <Link to="/quote">Quote</Link>
          </li>
        </ul>
      </nav>
    </>
  );
}
