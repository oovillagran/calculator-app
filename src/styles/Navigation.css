.nav-item {
  color: #fff;
  text-decoration: none;
  padding-left: 1rem;
}

@media only screen and (max-width: 768px) {
  .brand {
    display: none;
  }

  .nav-container {
    text-align: center;
    margin: auto;
  }
}
