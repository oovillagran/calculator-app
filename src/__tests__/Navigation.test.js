import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Navigation from '../components/Navigation';

describe('Navigation component', () => {
  test('Renders the navigation bar', () => {
    const { container } = render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>
    );
    const navBar = container.querySelector('.nav-bar');
    expect(navBar).toBeInTheDocument();
  });

  test('Renders the navigation bar using a snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Navigation />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
