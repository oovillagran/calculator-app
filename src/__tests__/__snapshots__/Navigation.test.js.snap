// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component Renders the navigation bar using a snapshot 1`] = `
<div>
  <nav
    class="nav-bar navbar navbar-expand-lg navbar-dark bg-dark"
  >
    <div
      class="container"
    >
      <a
        class="brand navbar-brand"
        href="/"
      >
        Math Magicians
      </a>
      <nav
        class="navbar navbar-expand navbar-light"
        id="responsive-navbar-nav"
      >
        <div
          class="me-auto navbar-nav"
        />
        <div
          class="nav-container navbar-nav"
        >
          <a
            class="nav-item"
            href="/"
          >
            Home
          </a>
          <a
            class="nav-item"
            href="/calculator"
          >
            Calculator
          </a>
          <a
            class="nav-item"
            href="/quote"
          >
            Quote
          </a>
        </div>
      </nav>
    </div>
  </nav>
</div>
`;
